File: backend/src/main/java/org/example/backend/model/WorkoutDay.java
package org.example.backend.model;

import java.time.DayOfWeek;
import java.util.List;
import java.util.Set;

public record WorkoutDay(String id, DayOfWeek day, WorkoutDayType type, Set<MuscleGroup> targetMuscles,
                         List<Exercise> exercises) {

   /* public WorkoutDay {
        targetMuscles= targetMuscles == null ? Set.of() : Set.copyOf(targetMuscles);
        exercises = exercises == null ? List.of() : List.copyOf(exercises);

        if (type == WorkoutDayType.REST) {
            targetMuscles = Set.of();
            exercises = List.of();
        } else {
            boolean targetCheck = targetMuscles.stream()
                    .allMatch(muscleGroup -> muscleGroup.getCategory() == type);
            if (!targetCheck) throw new IllegalArgumentException("All target muscles must match " + type);
        }
    }*/
}

File: frontend/src/types/WorkoutDay.ts
import type {Exercise} from "./Exercise.ts";
import type {DayOfWeek} from "./DayOfWeek.ts";
import type {WorkoutDayType} from "./WorkoutDayType.ts";

export type WorkoutDay = {
    id: string,
    day: DayOfWeek,
    type: WorkoutDayType,
    targetMuscles: string[],
    exercises: Exercise[];
}